server/command_hooks.go

```diff
@@ -509,3 +521,60 @@ func (p *Plugin) crash() {
 	y := 0
 	_ = 1 / y
 }
+
+func (p *Plugin) executeCommandListFiles(args *model.CommandArgs) *model.CommandResponse {
+	fileInfos, err := p.API.GetFileInfos(0, 10, &model.GetFileInfosOptions{
+		ChannelIds:     []string{args.ChannelId},
+		SortDescending: true,
+	})
+	if err != nil {
+		errorMessage := "Failed to get file list"
+		p.API.LogError(errorMessage, "err", err.Error())
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         errorMessage,
+		}
+	}
+
+	team, err := p.API.GetTeam(args.TeamId)
+	if err != nil {
+		errorMessage := "Failed to get team name"
+		p.API.LogError(errorMessage, "err", err.Error())
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         errorMessage,
+		}
+	}
+
+	permaLink := args.SiteURL + "/" + team.Name + "/pl/"
+	attachments := make([]*model.SlackAttachment, 0, len(fileInfos))
+	for _, f := range fileInfos {
+		user, userErr := p.API.GetUser(f.CreatorId)
+		if userErr != nil {
+			errorMessage := "Failed to get username"
+			p.API.LogError(errorMessage, "err", userErr.Error())
+			return &model.CommandResponse{
+				ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+				Text:         errorMessage,
+			}
+		}
+		attachments = append(attachments,
+			&model.SlackAttachment{
+				Title:     f.Name,
+				TitleLink: permaLink + f.PostId,
```

Comment:
this is really cool! It'll be great to include this file's public url as well. As I imagine most people using this api would want to display it's public url, and looking at `FileInfo` struct it's not very clear how would one go about doing that, so it'll be great to include an example here.

Reply:
Do you mean in addition to the link to the post Id or instead of the link to the post id?

Reply:
in addition works, looks like you already made the change, excellent! 👌